docs/summer-of-code-ideas/summer-of-code-reports/gsoc-2020-nipungarg.md

```diff
@@ -0,0 +1,348 @@
+
+Since none of the existing samples that we found had implemented this method, we decided to implement the new Overlapping Frames Sample.
+
+This sample renders a triangle with a post-processing effect split into multiple
+interleaved render passes.
+
+The idea was to create a sample that can be used to test the Overlapping Frames Synchronization pattern on the AGI. This pattern is described in more detail below.
+
+##### Frame Creation Pattern
+
+We have two render passes:
+- Geometry Render Pass (GRP)
+- Post Processing Render Pass (PPRP)
+
+For a given frame number N, the sample is submitting work as follows:
+
+... -> GRP\_{N} -> PPRP\_{N-1} -> GRP\_{N+1} -> PPRP\_{N} -> ...
+
+We use three sets of VkSemaphores and one set of VkFences to accomplish the necessary Vulkan
+synchronization. Each set has a size equal to the number of swapchain images
+(triple-buffered at a minimum).
+
+##### Semaphores
+- gRenderFinished: Signaled when the Geometry Render Pass is completed.
+- imageAcquired: Signaled when a swapchain image is acquired for the Post Processing Render Pass.
+- postRenderFinished: Signaled when the Post Processing Render Pass is completed.
+
+##### Fences
+- renderingFence: Signaled when the Post Processing Render Pass is completed.
+
+We use only one set of fences to synchronize both render passes. Since the two
+render passes are sequential - PPRP\_{N} cannot run before GRP\_{N} has been
+completed.
+
+PRs:
+[#169](https://github.com/google/vulkan_test_applications/pull/169):
+This PR creates a simple overlapping frames sample where we first draw a triangle to a temporary Vulkan Image, and then invert its color in separate render passes.
+
+[#184](https://github.com/google/vulkan_test_applications/pull/184):
+This PR modifies the overlapping_frames sample for different image each frame.
+
+### Vulkan Test Applications - Combining Graphics and Compute with Raymarching and Noise Texture
+The idea behind this sample was to combine the Compute and Graphics capabilities of Vulkan in order to visualize 3D Noise Textures. For the sample, we would procedurally generate a 3D Perlin Texture using the Compute queue and then use the resulting texture and a given surface level as inputs for a Raymarching Renderer.
+
+This was the final sample that was planned under the project. However, I could not get it to work satisfactorily under the current schedule. Currently, the setup for Graphics of the project exists. However, the shaders need to be re-written as they do not work as desired right now. The Compute side of the project is unimplemented till now.
```

Comment:
Replace the second "However" by "Still, ..."

`... till now` -> `... at the time of writing this report`.